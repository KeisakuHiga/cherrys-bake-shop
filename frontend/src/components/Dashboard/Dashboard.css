.DashBoard-container {
  width: 80vw;
}