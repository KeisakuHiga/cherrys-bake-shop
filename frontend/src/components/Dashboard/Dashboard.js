import React, { Component } from 'react'
import Moment from 'react-moment'
import style from './Dashboard.module.css'
import _ from 'lodash';
import TableBody from './TableBody/TableBody';
import DropDown from './DropDown/DropDown';
const axios = require('axios')

class Dashboard extends Component {
  state = {
    allQuotes: [],
    orderBy: "firstName",
    order: "asc",  // or "desc",
    dropDownActive: false

  }
  
  componentDidMount = () =>{
    this.getAllQuotes()
  }

  getAllQuotes = async () => {
    const token = localStorage.getItem('token')
    const url = `${process.env.REACT_APP_API_URL}/quote/getAllQuotes`
    const response = await axios.get(url, {headers: { token } });
    const data = await response.data;
    this.setState({
      allQuotes: data
    });
  };
  toggle = (e) => {
    e.preventDefault();
    let isActive = this.state.dropDownActive;
    isActive = !isActive;
    this.setState({dropDownActive: isActive});
  }
  doOrderBy = (e) => {
    e.preventDefault();
    const newOrderBy = e.target.getAttribute('data-value');
    this.setState({orderBy : newOrderBy});
  }
  doOrder = (e) => {
    e.preventDefault();
    const newOrder = e.target.getAttribute('data-value');
    this.setState({order : newOrder});
  }
  render() {
    let { orderBy, order, allQuotes }  = this.state
    // allQuotes  = _.orderBy(allQuotes, (item) => {
    //   return item[orderBy]
    // }, order)
    allQuotes  = _.orderBy(allQuotes,[orderBy], order)
    
    const items = allQuotes.map((item, index)=>{
      return <TableBody data={ item }
                        key={ item._id }
                        orderBy={ this.state.orderBy }
                        id={item.index}
                        />
    }); 

    if(!allQuotes) {
      return null
    } else {
      return (
        <>
          <h1 className={style.dashtitle}>All Quotes</h1>
          <div className={style.sortbutton}>
            <DropDown toggle={ this.toggle } 
                  dropDownActive={ this.state.dropDownActive } 
                  doOrderBy={ this.doOrderBy }
                  doOrder={ this.doOrder }
                  orderBy={ this.state.orderBy }
                  order={ this.state.order } />
          </div>
          <div className={style.dashboardcontainer}>
            <table className="table table-hover">
              <thead>
                <tr>
                  <th scope="col" >Number</th>
                  <th scope="col" >First Name</th>
                  <th scope="col" >Last Name</th>
                  <th scope="col" >Phone Number</th>
                  <th scope="col" >Estimated Pick Up</th>
                  <th scope="col" >Created At</th>
                </tr>
              </thead>
              <tbody>
                { items }
                {/* {allQuotes.map((quote, index) => (
                  <tr key={index} onClick={() => window.location =`/QuoteDetail/${quote._id}`} className={style.tablerow}>
                    <th key={quote._id} scope="row">{index + 1}</th>
                    <td key={quote.user.userName.firstName}>{quote.user.userName.firstName}</td>
                    <td key={quote.user.userName.lastName}>{quote.user.userName.lastName}</td>
                    <td key={quote.user.contact.phoneNumber}>{quote.user.contact.phoneNumber}</td>
                    <td key={quote.pickUp.time}><Moment local format="MMM DD, YYYY LT">{quote.pickUp.time}</Moment></td>
                    <td key={quote.createdAt}><Moment local format="MMM DD, YYYY LT">{quote.createdAt}</Moment></td>
                    </tr>
                ))} */}
              </tbody>
            </table>
          </div>
        </>
      )
    }
  }
}

export default Dashboard;


