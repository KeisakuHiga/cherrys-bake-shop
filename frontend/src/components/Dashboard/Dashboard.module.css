/* desktop design */

.dashtitle {
  display: flex;
  justify-content: center;
  align-items: center;
  font-size: 40px;
  margin-left: 8vw;
  width: 84vw;
  border-bottom: 2px solid rgba(0,0,0,0.10);
}

.dashboardcontainer{
  margin: 4% 8% 8% 8% ;
}

.tablerow {
  cursor: pointer;
}

.sortcont {
  width: 100vw;
  display: flex;
  justify-content: center;
  align-items: center;
}

.sortbutton {
  display: block;
}

/*  */
/* body {
  position: relative;
  padding-top: 50px;
  padding-bottom: 50px;
  font-size: 16px;  
}
body:before {
  position: absolute;
  top: 0;
  content: "";
  width: 100%;
  height: 100%;
  opacity: .5;
  background: url(http://m.rgbimg.com/cache1wmVD5/users/x/xy/xymonau/600/oy5PsrQ.jpg);
}
h1 {
  margin-bottom: 50px;
}
.col-sm-8 {
  background: #fff;
  border: 1px solid #8b7ecd;
  border-radius: 3px;
}
small {
  color: #5bc0de;
}
span.active {
  color: #9C27B0;
}
small.active {
  color: #9C27B0;
  font-weight: 700;
}
h4 {
  font-size: 20px;
}
.btn .caret {
  margin-left: 5px;
}
.author {
  padding: 50px 0 0;
  color: #9C27B0;
}
.media-left img {
  border: 1px solid #9C27B0;
  border-radius: 3px;
}
.dropdown-menu a {
  position: relative;
}
.glyphicon-ok {
  position: absolute;
  top: 4px;
  right: 15px;
} */