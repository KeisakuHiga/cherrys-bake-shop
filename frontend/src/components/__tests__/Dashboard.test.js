import React from "react";
import Enzyme, { shallow, render, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { createSerializer } from "enzyme-to-json";
import sinon, { spy } from "sinon";

// Set the default serializer for Jest to be the from enzyme-to-json
// This produces an easier to read (for humans) serialized format.
expect.addSnapshotSerializer(createSerializer({ mode: "deep" }));

// React 16 Enzyme adapter
Enzyme.configure({ adapter: new Adapter() });

// import chai, { expect } from "chai";
// import sinonChai from "sinon-chai";
// chai.use(sinonChai)

import Dashboard from "../Dashboard/Dashboard";

describe("<Dashboard />", () => {

  it('test', () => {
    const wrapper = render(
      <Dashboard />
    )
    expect(wrapper).toMatchSnapshot()
  })
});