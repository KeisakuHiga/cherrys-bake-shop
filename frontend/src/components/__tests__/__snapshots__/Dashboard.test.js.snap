// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dashboard /> test 1`] = `
<div
  class="dashcont"
>
  <h1
    class="dashtitle"
  >
    All Quotes
  </h1>
  <div
    class="sortbutton"
  >
    <select>
      <option
        hidden=""
        selected=""
        value=""
      >
        Sort By Category
      </option>
      <option
        value="user.userName.firstName"
      >
        First Name
      </option>
      <option
        value="user.userName.lastName"
      >
        Last Name
      </option>
      <option
        value="user.contact.phoneNumber"
      >
        Phone Number
      </option>
      <option
        value="pickUpDateAndTime"
      >
        Estimated Pick Up
      </option>
      <option
        value="createdAt"
      >
        Created At
      </option>
    </select>
    <select>
      <option
        hidden=""
        selected=""
        value=""
      >
        Sort Order
      </option>
      <option
        value="asc"
      >
        Ascending
      </option>
      <option
        value="desc"
      >
        Descending
      </option>
    </select>
  </div>
  <div
    class="dashboardcontainer"
  >
    <table
      class="table table-hover"
    >
      <thead>
        <tr>
          <th
            scope="col"
          >
            Number
          </th>
          <th
            scope="col"
          >
            First Name
          </th>
          <th
            scope="col"
          >
            Last Name
          </th>
          <th
            scope="col"
          >
            Phone Number
          </th>
          <th
            scope="col"
          >
            Estimated Pick Up
          </th>
          <th
            scope="col"
          >
            Created At
          </th>
        </tr>
      </thead>
      <tbody />
    </table>
  </div>
</div>
`;
